src/org/ssgwt/client/ui/form/spinner/ValueSpinner.java

```diff
@@ -64,50 +64,114 @@
+            // Set the new value for the valueBox
+            valueBox.setText(updateValue);
+            if (updateValue != null && !updateValue.equals("")) {
+                // Replace old value with a new one
+                oldValue = Long.parseLong(updateValue);
+            }
         }
     };
 
-    private final KeyPressHandler keyPressHandler = new KeyPressHandler() {
+    /**
+     * This is a blur handler for the numeric field
+     * 
+     * @author Michael Barnard <michael.barnard@a24group.com>
+     * @since  03 August 2015
+     */
+    private final BlurHandler blurHandler = new BlurHandler() {
 
+        /**
+         * This method will fire as soon as the spinner loses focus
+         * 
+         * @author Michael Barnard <michael.barnard@a24group.com>
+         * @since  27 July 2015
+         * 
+         * @param event - The blur event for the focus lost
+         */
         @Override
-        public void onKeyPress(KeyPressEvent event) {
-            int index = valueBox.getCursorPos();
-            String previousText = valueBox.getText();
-            String newText;
-            if (valueBox.getSelectionLength() > 0) {
-                newText = previousText.substring(0, valueBox.getCursorPos())
-                        + event.getCharCode()
-                        + previousText.substring(valueBox.getCursorPos()
-                                + valueBox.getSelectionLength(),
-                                previousText.length());
-            } else {
-                newText = previousText.substring(0, index)
-                        + event.getCharCode()
-                        + previousText.substring(index, previousText.length());
-            }
-            valueBox.cancelKey();
-            try {
-                long newValue = parseValue(newText);
-                if (spinner.isConstrained()
-                        && (newValue > spinner.getMax() || newValue < spinner
-                                .getMin())) {
-                    return;
+        public void onBlur(BlurEvent event) {
+            // Get the value from the text box
+            String value = valueBox.getValue();
+            if (!value.equals(formatValue(getSpinner().getValue()))){
+                try {
+                    // Convert text to number
+                    long numericNew = Long.parseLong(value);
+                    // Set minimum as 0
+                    if (numericNew < 0) {
+                        numericNew = 0;
+                    }
+                    
+                    // Calculate the difference in with the multiplier
+                    long difference = numericNew - oldValue;
+                    long timeDiff = difference * multiplier;
+                    long newValue = spinner.getValue() + timeDiff;
+                    
+                    // Get a limited value for the 
+                    long shortNewValue = Long.parseLong(formatValue(newValue));
```

Comment:
`shortNewValue` please update the name of this variable, since short is a valid type https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html